activerecord/lib/active_record/associations/builder/has_many.rb

```diff
@@ -7,14 +7,15 @@ def self.macro
     end
 
     def self.valid_options(options)
-      valid = super + [:counter_cache, :join_table, :index_errors]
+      valid = super + [:counter_cache, :join_table, :index_errors, :ensuring_owner_was]
```

Comment:
As @vendethiel said, here the key is always included, which invalidates line 13, no?

Reply:
As @adrianna-chang-shopify said, feel free to open a PR 🙂

Reply:
[Here](https://github.com/rails/rails/pull/40655) @eugeneius , could you review? @adrianna-chang-shopify